elastic/kibana
# [Saved Searches] Add support for saved searches by value

## Summary

This PR adds support for saved searches by value. Functionality-wise this means that similar to other Dashboard panels, saved search panels now allow unlinking from the library as well as cloning by value instead of by reference.

Testing guide:
- Test saved searches using both persisted and temporary data views.
- Ensure your saved searches include a query, filters, time range, columns, sorting, breakdown field, etc.
- Test both the "Unlink from library" and "Save to library" functionality.
- Test "Clone panel" functionality using both by reference and by value saved searches (both should clone to a by value saved search).
- Test the "Edit search" button functionality in Dashboard edit mode:
  - All saved search configurations should be included when navigating (search params + persisted & temporary data views).
  - Test navigation within the same tab (which will use in-app navigation to pass state) and opening the link in a new tab (which will use query params to pass state).
  - By reference saved searches should use the `/app/discover#/view/{savedSearchId}` route.
  - By value saved searches using persisted data views should pass all saved search configurations through the app state (`_a`) query param.
  - By value saved searches using temporary data views should use a locator redirect URL (`/app/r?l=DISCOVER_APP_LOCATOR...`) in order to support encoding their temporary data view in the URL state.
- Test the "Open in Discover" button functionality in Dashboard view mode:
  - By reference saved searches should open the actual saved search in Discover.
  - By value saved searches should pass all saved search configurations to Discover.

The following features are not included in this PR and comprise the remaining work for implementing Time to Visualize for saved searches (to be done at a later date; issue here: #141629):
- Save and return / state transfer service.
- Save modal with the ability to save directly to a dashboard.

Resolves #148995.
Unblocks #158632.

### Checklist


### For maintainers
